flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -187,8 +194,26 @@ componentToExtendedValue(Fortran::lower::FirOpBuilder &builder,
                          mlir::Location loc, const fir::ExtendedValue &obj,
                          mlir::Value component) {
   auto fieldTy = fir::dyn_cast_ptrEleTy(component.getType());
-  if (fieldTy.dyn_cast<fir::BoxType>())
-    TODO(loc, "lower pointer/allocatable component ref");
+  if (fieldTy.isa<fir::BoxType>()) {
+    llvm::SmallVector<mlir::Value> nonDeferredTypeParams;
+    auto eleTy = fir::unwrapSequenceType(fir::dyn_cast_ptrOrBoxEleTy(fieldTy));
+    if (auto charTy = eleTy.dyn_cast<fir::CharacterType>()) {
+      auto lenTy = builder.getCharacterLengthType();
+      if (charTy.hasConstantLen())
+        nonDeferredTypeParams.emplace_back(
+            builder.createIntegerConstant(loc, lenTy, charTy.getLen()));
+      // TODO: Starting, F2003, the dynamic character length might be non
+      // deferred (depending on derived type length parameters). There is no way
```

Comment:
Thanks, I did not realize deferred length was not F95. Good to know we are ahead of F95 on some topics (since deferred length allocatable/pointer are already lowered OK). I applied your suggestion.